docs/entities/index.md

```diff
@@ -0,0 +1,431 @@
+
+```java
+public MyEntity(Level level, double x, double y, double z) {
+    // Delegates to the factory constructor, using the EntityType we registered before.
+    this(MY_ENTITY.get(), level);
+    this.setPos(x, y, z);
+}
+```
+
+And now, we are free to do basically whatever we want with our entity. The following subsections will display a variety of common entity use cases.
+
+### Data Storage on Entities
+
+_See [Entities/Data and Networking][data]._
+
+### Rendering Entities
+
+_See [Entities/Entity Renderers][renderer]._
+
+### Spawning Entities
+
+If we now boot up the game now and enter a world, we have exactly one way of spawning: through the [`/summon`][summon] command (assuming `EntityType.Builder#noSummon` was not called).
+
+Obviously, we want to add our entities some other way. The easiest way to do so is through the `LevelWriter#addFreshEntity` method. This method simply accepts an `Entity` instance and adds it to the world, like so:
+
+```java
+// In some method that has a level available, only on the server
+if (!level.isClientSide()) {
+    MyEntity entity = new MyEntity(level, 100.0, 200.0, 300.0);
+    level.addFreshEntity(entity);
+}
+```
+
+This will be used for pretty much all non-living entities. Players should obviously not be spawned yourself, and `Mob`s have [their own ways of spawning][mobspawn] (though they can also be added via `#addFreshEntity`).
+
+### Damaging Entities
+
+_See also [Left-Clicking an Item][leftclick]._
+
+While not all entities have the concept of hit points, they can still all receive damage. This is not only used by things like mobs and players: If you cast your mind to item entities (dropped items), they too can take damage from sources like fire or cacti, in which case they are usually deleted immediately.
+
+Damaging an entity is possible by calling `Entity#hurt`. `Entity#hurt` takes two arguments: the [`DamageSource`][damagesource] and the damage amount, as a float in half hearts. For example, calling `entity.hurt(entity.damageSources().wither(), 4.25)` will cause a little over two hearts of wither damage.
+
+In turn, entities can also modify that behavior. This isn't done by overriding `#hurt`, as it is a final method. Rather, there are two methods `#hurtServer` and `#hurtClient` that each handle damage logic for the corresponding side. `#hurtClient` is commonly used to tell the client that an attack has succeeded, even though that may not always be true, mainly for playing attack sounds and other effects regardless. For changing damage behavior, we mainly care about `#hurtServer`, which we can override like so:
+
+```java
+@Override
+// The boolean return value determines whether the entity was actually damaged or not.
+public boolean hurtServer(ServerLevel level, DamageSource damageSource, float amount) {
+    if (damageSource.is(DamageTypeTags.IS_FIRE)) {
+        return super.hurt(level, damageSource, amount * 2);
```

Comment:
Not how this is implemented. Basically, it boils down to how the health is handled. For non-living, this is usually something custom, like setting entity damage for vehicles. Living entities set the health via `setHealth` usually called via `actuallyHurt`.